kin-sdk-core/src/main/java/kin/sdk/core/Request.java

```diff
@@ -0,0 +1,92 @@
+package kin.sdk.core;
+
+
+import android.os.Handler;
+import android.os.Looper;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+/**
+ * Represents {@link KinAccount} method invocation, each request will run sequentially on background thread,
+ * and will notify {@link ResultCallback} witch success or error on main thread.
+ *
+ * @param <T> request result type
+ */
+public class Request<T> {
+
+    private static final ExecutorService executorService = Executors.newSingleThreadExecutor();
+    private final Handler mainHandler;
+    private final Callable<T> callable;
+    private boolean cancelled;
+    private boolean executed;
+    private Future<?> future;
+    private ResultCallback<T> resultCallback;
+
+    Request(Callable<T> callable) {
+        checkNotNull(callable, "callable");
+        this.callable = callable;
+        this.mainHandler = new Handler(Looper.getMainLooper());
+    }
+
+    /**
+     * Run request asynchronously, notify {@code callback} with successful result or error
+     */
+    synchronized public void run(ResultCallback<T> callback) {
+        checkBeforeRun(callback);
+        executed = true;
+        submitFuture(callable, callback);
+    }
+
+    private void checkBeforeRun(ResultCallback<T> callback) {
+        checkNotNull(callback, "callback");
+        if (executed) {
```

Comment:
we could set executed to false after execution has ended - so that the same request could be re-used. WDYT?

Reply:
Don't think it's worth the effort, as we're not exposed `isExecuted`/`isCancelled`, there is no reliable use from client perspective. also I think it's a common property that tasks are for single use.